packages/wdio-cli/src/run.js

```diff
@@ -64,9 +64,24 @@ export default function run (params) {
     })
 }
 
+/**
+ * launcher callback function
+ * @param {number|undefined}    exitCode    exit code
+ * @param {Error|undefined}     err         Error
+ * @param {boolean|undefined}   isComplete  if all tests and hooks were executed
+ */
+const launchCallback = (exitCode, err, isComplete) => {
+    if (!err) {
+        return { exitCode }
+    }
+    console.error(err)
+    if (!isComplete) {
+        process.emit('shutdown', 1)
```

Comment:
if an exception was thrown before tests and hooks were completed we need to send `shutdown` event to end process normally. 

It should be `shutdown`, not `end` because we use [async-exit-hook](https://www.npmjs.com/package/async-exit-hook) to do something in the very end.